### PR TITLE
GL-2039: Added prompt for nodejs and php project selection. 

### PR DESCRIPTION
**Pull request by akashkska**
Codestudio is introducing feature to create node js project separately, hence implemented prompt to choose type of project and respective flow for project creation.

### CODE REVIEW

**Review by shriacquia — CHANGES_REQUESTED** [review]
Need to address mutant complaints.

File: src/Command/CodeStudio/CodeStudioWizardCommand.php [hunk 1]
```diff
@@ -42,14 +42,33 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+    $nodeVersion = NULL;
+    $projectType = [
+      'Drupal_project',
+      'Node_project',
     ];
-    $project = $this->io->choice('Select a PHP version', array_values($phpVersions), $phpVersions['PHP_version_8.1']);
-    $project = array_search($project, $phpVersions, TRUE);
-    $phpVersion = $phpVersions[$project];
+    $projectSelected = $this->io->choice('Select a project type', $projectType, $projectType[0]);
+    echo "Selected project is, $projectSelected";
```
Comment (danepowell) [inline src/Command/CodeStudio/CodeStudioWizardCommand.php:52]: Is this is a debugging statement?

File: src/Command/CodeStudio/CodeStudioWizardCommand.php [hunk 2]
```diff
@@ -163,9 +187,34 @@ private function createProjectAccessToken(array $project, string $projectAccessT
     return $projectAccessToken['token'];
   }
 
-  private function setGitLabCiCdVariables(array $project, string $cloudApplicationUuid, string $cloudKey, string $cloudSecret, string $projectAccessTokenName, string $projectAccessToken, string $phpVersion): void {
+  private function setGitLabCiCdVariablesForPhpProject(array $project, string $cloudApplicationUuid, string $cloudKey, string $cloudSecret, string $projectAccessTokenName, string $projectAccessToken, string $phpVersion): void {
+    $this->io->writeln("Setting GitLab CI/CD variables for {$project['path_with_namespace']}..");
+    $gitlabCicdVariables = CodeStudioCiCdVariables::getDefaultsForPhp($cloudApplicationUuid, $cloudKey, $cloudSecret, $projectAccessTokenName, $projectAccessToken, $phpVersion);
```
Comment (danepowell) [inline src/Command/CodeStudio/CodeStudioWizardCommand.php:192]: Don't rely on classes from the system under test (e.g., CodeStudioCiCdVariables). These variables should be hardcoded in the test case or fixture.

File: src/Command/CodeStudio/CodeStudioWizardCommand.php [hunk 3]
```diff
@@ -42,14 +42,33 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
     // So we reauthenticate to be sure we're using the provided credentials.
     $this->reAuthenticate($cloudKey, $cloudSecret, $this->cloudCredentials->getBaseUri(), $this->cloudCredentials->getAccountsUri());
-
-    $phpVersions = [
-      'PHP_version_8.1' => "8.1",
-      'PHP_version_8.2' => "8.2",
+    $phpVersion = NULL;
+    $nodeVersion = NULL;
+    $projectType = [
+      'Drupal_project',
```
Comment (danepowell) [inline src/Command/CodeStudio/CodeStudioWizardCommand.php:48]: Please kill the escaped mutants. For instance, this escaped mutant is telling you that if `Drupal_project` is removed as an available project type, tests still pass. That seems like something tests should catch..

**Comment by anavarre** [comment]
@akashkska action item on you. Will you be able to work on the PR in the current sprint?

**Comment by akashkska** [comment]
> @akashkska action item on you. Will you be able to work on the PR in the current sprint?

yes working on same. There are some understanding gaps related to mutation testing so it is taking time to resolve .

File: src/Command/CodeStudio/CodeStudioWizardCommand.php (hunk 1 above)
Comment (akashkska) [reply src/Command/CodeStudio/CodeStudioWizardCommand.php]: Removed debugging statement : [60bf6c6](https://github.com/acquia/cli/pull/1683/commits/60bf6c67023bcee873512fae1c88a952174a9347)

**Comment by danepowell** [comment]
To get you started, I recorded myself killing one of the mutants. Try replicating this process to kill a few more.

https://github.com/acquia/cli/assets/1984514/d80aef58-beaf-4449-9a24-05c6153c38ed

File: src/Command/CodeStudio/CodeStudioCiCdVariables.php [hunk 4]
```diff
@@ -10,13 +10,64 @@ class CodeStudioCiCdVariables {
...
+    return array_column(self::getDefaultsForPhp(), 'key');
   }
 
   /**
-   * @return array<mixed>
-   */
-  public static function getDefaults(?string $cloudApplicationUuid = NULL, ?string $cloudKey = NULL, ?string $cloudSecret = NULL, ?string $projectAccessTokenName = NULL, ?string $projectAccessToken = NULL, ?string $phpVersion = NULL): array {
+  * @return array<mixed>
+  */
+  public static function getDefaultsForNode(?string $cloudApplicationUuid = NULL, ?string $cloudKey = NULL, ?string $cloudSecret = NULL, ?string $projectAccessTokenName = NULL, ?string $projectAccessToken = NULL, ?string $nodeVersion = NULL): array {
```
Comment (danepowell) [inline src/Command/CodeStudio/CodeStudioCiCdVariables.php:20]: Isn't this basically the same as getMockGitlabVariables (with my additions)? Seems like they could be combined.

Comment (akashkska) [reply src/Command/CodeStudio/CodeStudioCiCdVariables.php:20]: It is almost same but there is change of one last variable. Hence we have introduce `getDefaultsForNode` and `getDefaultsForPhp` removing `getDefaults `

**Comment by akashkska** [comment]
> To get you started, I recorded myself killing one of the mutants. Try replicating this process to kill a few more. You don't need to kill 100% of them, but get as many as you can.
> 
>  Screen.Recording.2024-02-21.at.3.39.17.PM.mov

Is this correct recording ? I can see 3 sec recording.

**Comment by danepowell** [comment]
Sorry, indeed it was the wrong recording. I reuploaded it.

**Comment by akashkska** [comment]
@danepowell After latest commit No mutation warnings are shown also I see 2 mutants were killed.
So does that mean PR is good to go in main ?
If yes will you please approve and merge.

**Comment by danepowell** [comment]
Sorry, sometimes the tests time out. I reran them and will increase the timeout in `master`.

I think you'd only need to add 2-3 assertions to kill all or almost all of the mutants, like I did in the video. Please try it. It's not just busy work; the escaped mutants indicate the tests are not adequate, and I need to be able to rely on the tests in order to maintain this code.

**Comment by akashkska** [comment]
Apologise @danepowell fix might be silly, to add valid assertions. But for me its taking time due to lack of expertise on php, as this is first time I am interacting with php codebase and testcases.

**Comment by anavarre** [comment]
Dane is back at work today and will be able to further guide you on mutation testing. Good work thus far on addressing them.

**Review by danepowell — APPROVED** [review]

**Pull request merged**